File: src/main/java/chloe/movietalk/domain/MovieActor.java
package chloe.movietalk.domain;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Getter
@Table(name = "MOVIE_ACTOR")
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class MovieActor extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "movie_id")
    private Movie movie;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "actor_id")
    private Actor actor;

    public MovieActor(Movie movie, Actor actor) {
        this.movie = movie;
        this.actor = actor;
    }
}

File: src/main/java/chloe/movietalk/domain/MovieActor.kt
package chloe.movietalk.domain

import jakarta.persistence.*

@Entity
@Table(name = "MOVIE_ACTOR")
class MovieActor(
    @JoinColumn(name = "movie_id")
    @ManyToOne(fetch = FetchType.LAZY)
    var movie: Movie,

    @JoinColumn(name = "actor_id")
    @ManyToOne(fetch = FetchType.LAZY)
    var actor: Actor
) : BaseEntity() {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null
}
